terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/main/java/org/terasoluna/gfw/common/codelist/i18n/SimpleI18nCodeList.java

```diff
@@ -226,14 +232,54 @@
 
     /**
      * <p>
-     * returns row of codelist table.
+     * Returns a codelist as map for the {@link org.springframework.context.i18n.LocaleContextHolder}'s locale.
+     * </p>
+     * @see org.terasoluna.gfw.common.codelist.CodeList#asMap()
+     */
+    @Override
+    public Map<String, String> asMap() {
+        return asMap(LocaleContextHolder.getLocale());
+    }
+
+    /**
+     * <p>
+     * returns row of codelist table.<br>
+     * if there is no codelist for the locale, throws an {@link IllegalStateException}.
      * </p>
      * @see org.terasoluna.gfw.common.codelist.i18n.I18nCodeList#asMap(java.util.Locale)
      */
     @Override
     public Map<String, String> asMap(Locale locale) {
         Assert.notNull(locale, "locale is null");
-        return codeListTable.row(locale);
+
+        if (codeListTable.containsRow(locale)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Found codelist for specified locale '{}'.",
+                        locale);
+            }
+            return codeListTable.row(locale);
+        }
+
+        Locale langOnlyLocale = new Locale(locale.getLanguage());
```

Comment:
- [ ] If a language locale was originally specified, it is no mean to get the language locale again.

Reply:
Modified by 0b4bd23